artisatomic/__init__.py

```diff
@@ -1012,6 +1023,8 @@ def integrand(nu):
 
 
 def check_forbidden(levela, levelb) -> bool:
+    if levela is None or levela.parity is None or levelb.parity is None:
+        return False
```

Comment:
Because of energy_levels = [None]
Which I copied from your code to read Tanaka's data. Without initialising the array with None the lowest level doesn't get written out. With it levela doesn't have attribute parity and this fails

Reply:
Ok, that's a relic of our 1-based indexing that we inherited from the existing artis file formats. However, if any of your transitions are referencing the dummy level in the 0 index, that suggests that there is an off-by-one error in the indexing of transitions.

Reply:
Should the transitions array have a dummy value initialised too then?

Reply:
No, we don't really keep track of a transition index, and all transitions get written out in write_transition_data(). write_adata() skips the 0-index when writing out the list of levels for each ion.

Reply:
It's important to manually verify that the first couple of levels and transitions are recorded correctly in the artis output files, because indexing errors are very easy to make.

Reply:
One file with the transitions is ~30MB

Reply:
Just take a sample of a few hundred transitions and compress it with zstd.